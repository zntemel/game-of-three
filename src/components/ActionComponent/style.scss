.action-component {
  display: flex;
  button {
    border: none;
    border-radius: 10px;
    height: 50px;
    width: 50px;
    margin-right: 30px;
    background: #f7637b;
    color: #fff;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 18px;
    font-family: "Kanit", sans-serif;
  }
}
