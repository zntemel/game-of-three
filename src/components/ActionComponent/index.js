import React from "react";
import PropTypes from "prop-types";
import { Button } from "antd";
import "./style.scss";

function ActionComponent(props) {
  return (
    <div className="action-component">
      <Button
        value={-1}
        onClick={() => props.actionValue(-1)}
        disabled={props.isButtonDisable}
      >
        -1
      </Button>
      <Button
        value={0}
        onClick={() => props.actionValue(0)}
        disabled={props.isButtonDisable}
      >
        <span>0</span>
      </Button>
      <Button
        value={1}
        onClick={() => props.actionValue(1)}
        disabled={props.isButtonDisable}
      >
        <span>1</span>
      </Button>
    </div>
  );
}

ActionComponent.propTypes = {
  actionValue: PropTypes.func,
  isButtonDisable: PropTypes.bool,
};

ActionComponent.defaultProps = {
  isButtonDisable: false,
};

export default ActionComponent;
