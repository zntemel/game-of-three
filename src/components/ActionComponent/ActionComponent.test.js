import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import ActionComponent from "./index";

describe("ActionComponent", () => {
  const wrapper = shallow(<ActionComponent />);
  const instance = wrapper.instance();

  describe("shallow wrapper", () => {
    it("should render succesfully with snapshots", () => {
      const component = renderer.create(wrapper).toJSON();
      expect(component).toMatchSnapshot();
    });

    it("instance should be null because of it's dumy component", () => {
      expect(instance).toEqual(null);
    });

    it("should be include 3 button", () => {
      expect(wrapper.find("Button")).toHaveLength(3);
    });
  });
});
