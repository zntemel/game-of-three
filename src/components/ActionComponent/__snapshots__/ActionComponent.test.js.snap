// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionComponent shallow wrapper should render succesfully with snapshots 1`] = `
<div
  className="action-component"
>
  <button
    className="ant-btn"
    disabled={false}
    onClick={[Function]}
    type="button"
    value={-1}
  >
    <span>
      -1
    </span>
  </button>
  <button
    className="ant-btn"
    disabled={false}
    onClick={[Function]}
    type="button"
    value={0}
  >
    <span>
      0
    </span>
  </button>
  <button
    className="ant-btn"
    disabled={false}
    onClick={[Function]}
    type="button"
    value={1}
  >
    <span>
      1
    </span>
  </button>
</div>
`;
